.github/workflows/integration-tests.yml

```diff
@@ -42,12 +42,12 @@ jobs:
           fi
 
   stateless-validator:
```

Comment:
I think this is already getting a bit complex, so I'm thinking on moving integration testing as Rust tests in a separate crate. 

Also because most of this integration testing would be also useful to run locally with a single command or similar, and not rely on CI.